kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/IcebergSinkConfig.java

```diff
@@ -333,10 +341,16 @@ public TableSinkConfig tableConfig(String tableName) {
           Pattern routeRegex = routeRegexStr == null ? null : Pattern.compile(routeRegexStr);
 
           String idColumnsStr = tableConfig.getOrDefault(ID_COLUMNS, tablesDefaultIdColumns());
+          if (schemaForceColumnsToLowercase() && idColumnsStr != null) {
+            idColumnsStr = idColumnsStr.toLowerCase();
```

Comment:
I feel we should remove this, it is defined in the config so a user can set this to lower case as needed.